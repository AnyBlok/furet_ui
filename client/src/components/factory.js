import Vue from 'vue';

const components = {};
const componentNames = [];

export const defineComponent = (elementName, declaration) => {
  let klass;
  if (elementName in components) {
    klass = components[elementName];
  } else {
    klass = {
      template: undefined,
      extend: [],
      prototypes: [],
    };
    components[elementName] = klass;
    componentNames.push(elementName);
  }
  if (declaration.template !== undefined) klass.template = declaration.template;
  if (declaration.prototype !== undefined) {
    klass.prototypes.push(declaration.prototype);
  }
  if (declaration.extend !== undefined) {
    declaration.extend.forEach((extend) => {
      klass.extend.push(extend);
    });
  }
};

window.defineComponent = defineComponent;

const getPrototypeFor = (elementName) => {
  const component = components[elementName];
  const prototypes = [];
  component.extend.forEach((otherElementName) => {
    prototypes.push(...getPrototypeFor(otherElementName));
  });
  prototypes.push(...component.prototypes);
  return prototypes;
};

export const createComponents = () => {
  componentNames.forEach((elementName) => {
    const component = components[elementName];
    const mixins = getPrototypeFor(elementName);
    Vue.component(elementName, (resolve) => {
      resolve({
        template: component.template,
        mixins,
      });
    });
  });
};
