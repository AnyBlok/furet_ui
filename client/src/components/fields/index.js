import './vue2-editor';
