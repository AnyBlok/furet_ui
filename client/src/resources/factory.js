import _ from 'underscore';

const resources = {};
export const resourceNames = [];

export const removeResource = (resourceName) => {
  const index = resourceNames.indexOf(resourceName);
  if (index !== -1) {
    delete resources[resourceName];
    resourceNames.splice(index, 1);
  }
};

export const defineResource = (resourceName, declaration) => {
  let klass;
  if (resourceName in resources) {
    klass = resources[resourceName];
  } else {
    klass = {
      children: [],
      name: resourceName,
      path: resourceName,
      templateName: resourceName,
      props: [],
      mustBeAuthenticated: false,
    };
    resources[resourceName] = klass;
    resourceNames.push(resourceName);
  }
  if (declaration.path !== undefined) klass.path = declaration.path;
  if (declaration.templateName !== undefined) klass.templateName = declaration.templateName;
  if (declaration.props !== undefined) klass.props = declaration.props;
  if (declaration.mustBeAuthenticated !== undefined) {
    klass.mustBeAuthenticated = declaration.mustBeAuthenticated;
  }
  if (declaration.children !== undefined) {
    declaration.children.forEach((child) => {
      let childDefinition = _.find(klass.children, c => c.name === child.name);
      if (childDefinition === undefined) {
        childDefinition = {
          name: child.name,
          completeName: `${resourceName}_${child.name}`,
          templateName: child.name,
          path: child.name,
          props: [],
          mustBeAuthenticated: false,
        };
        klass.children.push(childDefinition);
      }
      if (child.path !== undefined) childDefinition.path = child.path;
      if (child.templateName !== undefined) childDefinition.templateName = child.templateName;
      if (child.props !== undefined) childDefinition.props.push(...child.props);
      if (child.mustBeAuthenticated !== undefined) {
        childDefinition.mustBeAuthenticated = child.mustBeAuthenticated;
      }
    });
  }
};

window.defineResource = defineResource;

const createRoute = (resource, defaultTemplateProps) => {
  let path = `/${resource.path}`;
  let templateProps = defaultTemplateProps || '';
  if (resource.path === resource.name) {
    if (resource.props.length !== 0) {
      resource.props.forEach((prop) => {
        path += `/:${prop}`;
        templateProps += ` v-bind:${prop}="${prop}"`;
      });
    }
  } else if (resource.props.length !== 0) {
    resource.props.forEach((prop) => {
      templateProps += ` v-bind:${prop}="${prop}"`;
    });
  }
  const route = {
    name: resource.completeName || resource.name,
    path,
    props: templateProps !== '',
    meta: { requiresAuth: resource.mustBeAuthenticated },
    component: { props: resource.props },
  };
  if (resource.children === undefined || resource.children.length === 0) {
    route.component.template = `<${resource.templateName}${templateProps} />`;
  } else {
    route.children = [];
    if (resource.templateName === resource.name) {
      route.component.template = '<router-view />';
    } else {
      route.component.template = `<${resource.templateName}${templateProps} />`;
    }
    resource.children.forEach((child) => {
      route.children.push(createRoute(child, templateProps));
    });
  }
  return route;
};

export const getRoutes = () => {
  const routes = [];
  resourceNames.forEach((resourceName) => {
    const resource = resources[resourceName];
    routes.push(createRoute(resource));
  });
  return routes;
};
