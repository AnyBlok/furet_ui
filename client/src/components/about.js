import { defineComponent } from './factory';

defineComponent('about', {
  template: `
    <section class="section">
      <div class="container has-text-centered">
        <ul>
          <li><strong>version</strong> 0.0.0</li>
        </ul>
      </div>
    </section>
  `,
});
