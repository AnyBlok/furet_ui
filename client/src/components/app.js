import { defineComponent } from './factory';

defineComponent('app', {
  template: `
    <div id="furet-ui-app">
      <furet-ui-appbar />
      <notifications />
      <router-view></router-view>
      <furet-ui-footer />
    </div>`,
});

defineComponent('furet-ui-appbar', {
  template: `
    <header class="hero is-dark">
      <furet-ui-appbar-header />
      <furet-ui-appbar-body />
      <furet-ui-appbar-footer />
    </header>
  `,
});

defineComponent('furet-ui-appbar-header', {
  template: `
    <div class="hero-head">
      <nav class="navbar is-dark" role="navigation">
        <div class="container">
          <div class="navbar-brand">
            <furet-ui-appbar-header-brand />
            <a 
              role="button" 
              v-bind:class="['navbar-burger', 'burger', isOpen ? 'is-active' : '']" 
              aria-label="menu" 
              aria-expanded="false" 
              data-target="navbarUserMenu" 
              v-on:click="isOpen = !isOpen"
            >
              <span aria-hidden="true"></span>
              <span aria-hidden="true"></span>
              <span aria-hidden="true"></span>
            </a>
          </div>
          <div id="navbarUserMenu" v-bind:class="['navbar-menu', isOpen ? 'is-active' : '']">
            <furet-ui-appbar-spaces-menu />
            <furet-ui-appbar-user-menu />
          </div>
        </div>
      </nav>
    </div>
  `,
  prototype: {
    data() {
      return {
        isOpen: false,
      };
    },
  },
});

defineComponent('furet-ui-appbar-header-brand', {
  template: `
    <img class="image is-64x64" v-bind:src="logo" alt="Logo">
  `,
  extend: ['mixin-logo'],
});

defineComponent('furet-ui-appbar-body', {
  template: `
    <div class="hero-body">
      <div class="container has-text-centered">
        <h1 class="title">
          Furet UI
        </h1>
        <p class="subtitle">
          Web Client for any backend serveur
        </p>
      </div>
    </div>
  `,
});

defineComponent('furet-ui-appbar-footer', {
  template: `
    <div class="hero-foot" v-if="hasSpaceMenus">
      <furet-ui-appbar-space-menus />
    </div>
  `,
  prototype: {
    computed: {
      hasSpaceMenus() {
        return this.$store.state.menus.spaceMenus.length !== 0;
      },
    },
  },
});

defineComponent('furet-ui-appbar-user-menu', {
  prototype: {
    render(createElement) {
      const menus = [];
      this.menus.forEach((menu) => {
        menus.push(createElement(
          menu.component,
          {
            class: menu.class,
            style: menu.style,
            attrs: menu.attrs,
            props: menu.props,
            domProps: menu.domProps,
            on: menu.on,
            nativeOn: menu.nativeOn,
          },
          [menu.label]));
      });
      return createElement('div', { class: 'navbar-end' }, menus);
    },
    computed: {
      menus() {
        return this.$store.state.menus.user;
      },
    },
  },
});

defineComponent('furet-ui-appbar-router-link-goto', {
  prototype: {
    props: ['label', 'to'],
    methods: {
      goTo() {
        // TODO clear data
        this.$router.push(this.to);
      },
    },
  },
});

defineComponent('furet-ui-appbar-head-router-link', {
  template: `
    <a class="navbar-item" v-on:click="goTo">
      {{ label }}
    </a>
  `,
  extend: ['furet-ui-appbar-router-link-goto'],
});

defineComponent('furet-ui-appbar-head-router-link-button', {
  template: `
    <span class="navbar-item">
      <a class="button is-dark is-inverted is-fullwidth" v-on:click="goTo">
        <span class="icon" v-if="icon">
          <font-awesome-icon v-bind:icon="icon" />
        </span>
        <span>{{ label }}</span>
      </a>
    </span>
  `,
  extend: ['furet-ui-appbar-router-link-goto'],
  prototype: {
    props: ['icon'],
  },
});

defineComponent('furet-ui-appbar-spaces-menu', {
  prototype: {
    render(createElement) {
      const menus = [];
      this.menus.forEach((menu) => {
        menus.push(createElement(
          menu.component,
          {
            class: menu.class,
            style: menu.style,
            attrs: menu.attrs,
            props: menu.props,
            domProps: menu.domProps,
            on: menu.on,
            nativeOn: menu.nativeOn,
          },
          [menu.label]));
      });
      return createElement('div', { class: 'navbar-start' }, menus);
    },
    computed: {
      menus() {
        return this.$store.state.menus.spaces;
      },
    },
  },
});

defineComponent('furet-ui-appbar-space-menus', {
  prototype: {
    render(createElement) {
      const menus = [];
      this.menus.forEach((menu) => {
        menus.push(createElement('li', [
          createElement(
            menu.component,
            {
              class: menu.class,
              style: menu.style,
              attrs: menu.attrs,
              props: menu.props,
              domProps: menu.domProps,
              on: menu.on,
              nativeOn: menu.nativeOn,
            },
            [menu.label]),
        ]));
      });
      return createElement('nav', { class: 'tabs is-boxed' }, [
        createElement('div', { class: 'container' }, [
          createElement('ul', [menus]),
        ]),
      ]);
    },
    computed: {
      menus() {
        return this.$store.state.menus.spaceMenus;
      },
    },
  },
});

defineComponent('furet-ui-appbar-foot-router-link', {
  template: `
    <a v-on:click="goTo">
      {{ label }}
    </a>
  `,
  extend: ['furet-ui-appbar-router-link-goto'],
});

defineComponent('furet-ui-appbar-foot-router-link-button', {
  template: `
    <a class="button is-dark is-inverted" v-on:click="goTo">
      <span class="icon" v-if="icon">
        <font-awesome-icon v-bind:icon="icon" />
      </span>
      <span>{{ label }}</span>
    </a>
  `,
  extend: ['furet-ui-appbar-router-link-goto'],
  prototype: {
    props: ['icon'],
  },
});

defineComponent('furet-ui-footer', {
  template: `
    <footer class="footer">
      <div class="content has-text-centered is-success">
        <p>
          <strong>Furet UI</strong> by <a href="https://github.com/jssuzanne">Jean-SÃ©bastien Suzanne</a>. 
          The source code is licensed <a href="http://opensource.org/licenses/MPL-2.0">MPL-2.0</a> and 
          is available on <a href="https://github.com/AnyBlok/furet_ui">Github</a>.
        </p>
      </div>
    </footer>
  `,
});
