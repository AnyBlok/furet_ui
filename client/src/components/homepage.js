import { defineComponent } from './factory';

defineComponent('homepage', {
  template: `
    <div class="container has-text-centered">
      <h1 class="title">Welcome in Furet UI</h1>
      <div class="column is-half is-offset-one-quarter">
        <img class="image _s" v-bind:src="logo" alt="Logo">
      </div>
      <router-link to="/ping">Go to Ping</router-link>
    </div>
  `,
  extend: ['mixin-logo'],
});
