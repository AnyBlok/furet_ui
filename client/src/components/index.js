import './mixins';
import './app';
import './homepage';
import './login';
import './logout';
import './about';
import './list';

export { createComponents, default } from './factory';
