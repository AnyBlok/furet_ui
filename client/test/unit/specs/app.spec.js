import { mount } from 'vue-test-utils'
import '@/components';
import { getComponentPrototype } from '@/components/factory';
import sinon from 'sinon';


describe('app component', () => {
  const wrapper = mount(getComponentPrototype('app'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar component', () => {
  const wrapper = mount(getComponentPrototype('furet-ui-appbar'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-footer component', () => {
  const wrapper = mount(getComponentPrototype('furet-ui-footer'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-header component', () => {
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-header'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-header-brand component', () => {
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-header-brand'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-body component', () => {
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-body'));
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-footer component', () => {
  it('snapshot (empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaceMenus: []}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-footer'), {mocks});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('snapshot (not empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaceMenus: [
        {
          attrs: {},
          class: {},
          component: "furet-ui-appbar-foot-router-link",
          domProps: {},
          name: "test",
          nativeOn: {},
          on: {},
          props: {
            foo: "bar",
          },
          style: {},
        },
      ]}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-footer'), {mocks});

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-user-menu component', () => {
  it('snapshot (empty)', () => {
    const mocks = {
      $store: {state: {menus: {user: []}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-user-menu'), {mocks});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('snapshot (not empty)', () => {
    const mocks = {
      $store: {state: {menus: {user: [
        {
          attrs: {},
          class: {},
          component: "furet-ui-appbar-head-router-link",
          domProps: {},
          name: "test",
          nativeOn: {},
          on: {},
          props: {
            foo: "bar",
          },
          style: {},
        },
      ]}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-user-menu'), {mocks});

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-head-router-link component', () => {
  const spyRouter = sinon.spy();
  const mocks = {
    $router: {push: spyRouter},
  }
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-head-router-link'), {mocks});
  wrapper.setProps({
    label: 'fooBar',
    to: '/fooBar',
  });
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('click', () => {
    wrapper.find('a').trigger('click');
    expect(spyRouter.called).toBe(true);
    expect(spyRouter.firstCall.lastArg).toBe('/fooBar');
  });
});

describe('furet-ui-appbar-head-router-link-button component', () => {
  const spyRouter = sinon.spy();
  const mocks = {
    $router: {push: spyRouter},
  }
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-head-router-link-button'), {mocks});
  wrapper.setProps({
    label: 'fooBar',
    to: '/fooBar',
    icon: 'spinner',
  });
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('click', () => {
    wrapper.find('a').trigger('click');
    expect(spyRouter.called).toBe(true);
    expect(spyRouter.firstCall.lastArg).toBe('/fooBar');
  });
});

describe('furet-ui-appbar-spaces-menu component', () => {
  it('snapshot (empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaces: []}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-spaces-menu'), {mocks});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('snapshot (not empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaces: [
        {
          attrs: {},
          class: {},
          component: "furet-ui-appbar-head-router-link",
          domProps: {},
          name: "test",
          nativeOn: {},
          on: {},
          props: {
            foo: "bar",
          },
          style: {},
        },
      ]}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-spaces-menu'), {mocks});

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-space-menus component', () => {
  it('snapshot (empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaceMenus: []}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-space-menus'), {mocks});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('snapshot (not empty)', () => {
    const mocks = {
      $store: {state: {menus: {spaceMenus: [
        {
          attrs: {},
          class: {},
          component: "furet-ui-appbar-foot-router-link",
          domProps: {},
          name: "test",
          nativeOn: {},
          on: {},
          props: {
            foo: "bar",
          },
          style: {},
        },
      ]}}},
    }
    const wrapper = mount(getComponentPrototype('furet-ui-appbar-space-menus'), {mocks});

    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('furet-ui-appbar-foot-router-link component', () => {
  const spyRouter = sinon.spy();
  const mocks = {
    $router: {push: spyRouter},
  }
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-foot-router-link'), {mocks});
  wrapper.setProps({
    label: 'fooBar',
    to: '/fooBar',
  });
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('click', () => {
    wrapper.find('a').trigger('click');
    expect(spyRouter.called).toBe(true);
    expect(spyRouter.firstCall.lastArg).toBe('/fooBar');
  });
});

describe('furet-ui-appbar-foot-router-link-button component', () => {
  const spyRouter = sinon.spy();
  const mocks = {
    $router: {push: spyRouter},
  }
  const wrapper = mount(getComponentPrototype('furet-ui-appbar-foot-router-link-button'), {mocks});
  wrapper.setProps({
    label: 'fooBar',
    to: '/fooBar',
    icon: 'spinner',
  });
  it('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('click', () => {
    wrapper.find('a').trigger('click');
    expect(spyRouter.called).toBe(true);
    expect(spyRouter.firstCall.lastArg).toBe('/fooBar');
  });
});
