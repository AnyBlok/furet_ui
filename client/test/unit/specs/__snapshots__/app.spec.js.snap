// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component snapshot 1`] = `
<div
  id="furet-ui-app"
>
  <furet-ui-appbar />
   
  <notifications />
   
  <router-view />
   
  <furet-ui-footer />
</div>
`;

exports[`furet-ui-appbar component snapshot 1`] = `
<header
  class="hero is-dark"
>
  <furet-ui-appbar-header />
   
  <furet-ui-appbar-body />
   
  <furet-ui-appbar-footer />
</header>
`;

exports[`furet-ui-appbar-body component snapshot 1`] = `
<div
  class="hero-body"
>
  <div
    class="container has-text-centered"
  >
    <h1
      class="title"
    >
      
          Furet UI
        
    </h1>
     
    <p
      class="subtitle"
    >
      
          Web Client for any backend serveur
        
    </p>
  </div>
</div>
`;

exports[`furet-ui-appbar-foot-router-link component snapshot 1`] = `
<a>
  
      fooBar
    
</a>
`;

exports[`furet-ui-appbar-foot-router-link-button component snapshot 1`] = `
<a
  class="button is-dark is-inverted"
>
  <span
    class="icon"
  >
    <font-awesome-icon
      icon="spinner"
    />
  </span>
   
  <span>
    fooBar
  </span>
</a>
`;

exports[`furet-ui-appbar-footer component snapshot (empty) 1`] = `<!---->`;

exports[`furet-ui-appbar-footer component snapshot (not empty) 1`] = `
<div
  class="hero-foot"
>
  <furet-ui-appbar-space-menus />
</div>
`;

exports[`furet-ui-appbar-head-router-link component snapshot 1`] = `
<a
  class="navbar-item"
>
  
      fooBar
    
</a>
`;

exports[`furet-ui-appbar-head-router-link-button component snapshot 1`] = `
<span
  class="navbar-item"
>
  <a
    class="button is-dark is-inverted is-fullwidth"
  >
    <span
      class="icon"
    >
      <font-awesome-icon
        icon="spinner"
      />
    </span>
     
    <span>
      fooBar
    </span>
  </a>
</span>
`;

exports[`furet-ui-appbar-header component snapshot 1`] = `
<div
  class="hero-head"
>
  <nav
    class="navbar is-dark"
    role="navigation"
  >
    <div
      class="container"
    >
      <div
        class="navbar-brand"
      >
        <furet-ui-appbar-header-brand />
         
        <a
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger burger"
          data-target="navbarUserMenu"
          role="button"
        >
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
           
          <span
            aria-hidden="true"
          />
        </a>
      </div>
       
      <div
        class="navbar-menu"
        id="navbarUserMenu"
      >
        <furet-ui-appbar-spaces-menu />
         
        <furet-ui-appbar-user-menu />
      </div>
    </div>
  </nav>
</div>
`;

exports[`furet-ui-appbar-header-brand component snapshot 1`] = `
<img
  alt="Logo"
  class="image is-64x64"
  src="undefined/furet-ui/logo"
/>
`;

exports[`furet-ui-appbar-space-menus component snapshot (empty) 1`] = `
<nav
  class="tabs is-boxed"
>
  <div
    class="container"
  >
    <ul />
  </div>
</nav>
`;

exports[`furet-ui-appbar-space-menus component snapshot (not empty) 1`] = `
<nav
  class="tabs is-boxed"
>
  <div
    class="container"
  >
    <ul>
      <li>
        <furet-ui-appbar-foot-router-link
          class=""
        />
      </li>
    </ul>
  </div>
</nav>
`;

exports[`furet-ui-appbar-spaces-menu component snapshot (empty) 1`] = `
<div
  class="navbar-start"
/>
`;

exports[`furet-ui-appbar-spaces-menu component snapshot (not empty) 1`] = `
<div
  class="navbar-start"
>
  <furet-ui-appbar-head-router-link
    class=""
  />
</div>
`;

exports[`furet-ui-appbar-user-menu component snapshot (empty) 1`] = `
<div
  class="navbar-end"
/>
`;

exports[`furet-ui-appbar-user-menu component snapshot (not empty) 1`] = `
<div
  class="navbar-end"
>
  <furet-ui-appbar-head-router-link
    class=""
  />
</div>
`;

exports[`furet-ui-footer component snapshot 1`] = `
<footer
  class="footer"
>
  <div
    class="content has-text-centered is-success"
  >
    <p>
      <strong>
        Furet UI
      </strong>
       by 
      <a
        href="https://github.com/jssuzanne"
      >
        Jean-SÃ©bastien Suzanne
      </a>
      . 
          The source code is licensed 
      <a
        href="http://opensource.org/licenses/MPL-2.0"
      >
        MPL-2.0
      </a>
       and 
          is available on 
      <a
        href="https://github.com/AnyBlok/furet_ui"
      >
        Github
      </a>
      .
        
    </p>
  </div>
</footer>
`;
